### ISSUE TITLE
Sending non array causes UnexpectedTypeException

### ISSUE THREAD

**Issue by BruceWouaigne** [symfony/symfony#13804]
Hi guys,

I just notice that, when I send a non array data to a collection form field the `ResizeFormListener` throws an `UnexpectedTypeException` before the form validation.

Here is the action

```
public function indexAction(Request $request)
{
    $form = $this->get('form.factory')->createNamed('', new FooType, null, [
        'method' => 'GET'
    ]);

    $request->query->set('emails', 'a');

    // Throws an exception
    $form->handleRequest($request);

    if ($form->isValid()) {
        die(var_dump('valid'));
    }

    die(var_dump($form->getErrorsAsString()));
}

```

And the form

```
class FooType extends AbstractType
{
    public function buildForm(FormBuilderInterface $builder, array $options)
    {
        $builder->add('emails', 'collection', [
            'type' => 'email'
        ]);
    }

    public function getName()
    {
        return 'foo';
    }
}
```

Is there anything that I can do to avoid this? By this I mean to get a non valid form instead of an Exception...

I don't really get why https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Form/Extension/Core/EventListener/ResizeFormListener.php#L147 should throws an Exception instead of doing a `$data = [];` or something.

What do you think?

**Comment by jakzal** [comment]
I think the behaviour is correct at the moment. You're trying to submit a string instead of a collection. This is unexpected and an exception seems appropriate.

**Comment by Tobion** [comment]
The behavior is not correct when an exception is thrown. I already fixed some of those problems in the past where users can throw internal exceptions by submitting unexpected data. This should not be possible.

**Comment by Tobion** [comment]
See #5337 and #8282

**Comment by jakzal** [comment]
@Tobion makes sense. I missed your previous PRs.

**Comment by BruceWouaigne** [comment]
Alright then, should I create a pull request based on 2.3 or newer?

**Comment by BruceWouaigne** [comment]
Should I do something specific for someone merge the PR?

**Comment by xabbuh** [comment]
@BruceWouaigne Which PR do you refer to?

**Comment by webmozart** [comment]
Fixed in #13851.
